### PR TITLE
Update docs for slow start with priorities

### LINKED ISSUE #25283: slow-start not work with endpoints lb_endpoints.priority
*Title*: *slow-start not work with lb_endpoints.priority*

*Description*:
In production we use cluster with main/secondary endpoints. When main endpoint restarted, we expect that traffic follow to secondary endpoint until health check on main will not to be ok, and finally traffic should return to main endpoint with slow start. But slow start doesnot work in case with priority. Screens: 

When enable priority (on main endpoint:9000 priority = 0, on secondary:9001 priority = 1), slow start doesnt work as expected:
first endpoint (9000 port):
image
second endpoint (9001 port):
image

But when all priority the same - slow start works ok, but logic with main and second endpoints doesnt work:
first endpoint (9000 port):
image
second endpoint (9001 port):
image

*Config*:
envoy version: v1.25
config with priority:
```yaml
admin:
  address:
    socket_address:
      address: 0.0.0.0
      port_value: 9901

static_resources:
  listeners:
    - name: http
      address:
        socket_address:
          protocol: TCP
          address: 0.0.0.0
          port_value: 8080
      listener_filters:
        - name: envoy.filters.listener.proxy_protocol
          typed_config:
            "@type": type.googleapis.com/envoy.extensions.filters.listener.proxy_protocol.v3.ProxyProtocol
            allow_requests_without_proxy_protocol: true
      filter_chains:
        - filters:
            - name: envoy.filters.network.http_connection_manager
              typed_config:
                "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
                stat_prefix: http
                use_remote_address: true
                codec_type: AUTO
                normalize_path: true
                merge_slashes: true
                path_with_escaped_slashes_action: UNESCAPE_AND_REDIRECT
                common_http_protocol_options:
                  idle_timeout: 3600s
                  headers_with_underscores_action: REJECT_REQUEST
                http2_protocol_options:
                  max_concurrent_streams: 100
                  initial_stream_window_size: 65536
                  initial_connection_window_size: 1048576
                stream_idle_timeout: 300s
                request_timeout: 300s
                route_config:
                  name: local_route
                  virtual_hosts:
                    - name: backend
                      domains:
                        - '*'
                      routes:
                        - match:
                            prefix: "/"
                          route:
                            retry_policy:
                              host_selection_retry_max_attempts: 3
                              retry_on: 5xx,refused-stream,connect-failure
                              num_retries: 2
                            cluster: default
                http_filters:
                  - name: envoy.filters.http.router
                    typed_config:
                      "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
  clusters:
    - name: default
      type: STRICT_DNS
      outlier_detection:
        consecutive_5xx: 5
      connect_timeout: 5s
      lb_policy: ROUND_ROBIN
      health_checks:
        - timeout: 10s
          interval: 1s
          unhealthy_threshold: 1
          healthy_threshold: 1
          no_traffic_interval: 1s
          no_traffic_healthy_interval: 1s
          http_health_check:
            path: /target
      load_assignment:
        cluster_name: default
        endpoints:
          - lb_endpoints:
              - endpoint: #main
                  address:
                    socket_address:
                      address: 127.0.0.1
                      port_value: 9000
                load_balancing_weight: 100
          - lb_endpoints:
              - endpoint: #secondary
                  address:
                    socket_address:
                      address: 127.0.0.1
                      port_value: 9001
                load_balancing_weight: 100
            priority: 1
      round_robin_lb_config:
        slow_start_config:
          slow_start_window: 30s
          min_weight_percent: 
            value: 10
```

### PR DESCRIPTION
**Pull request by nezdolik**
Commit Message: Update user docs for slow start with priorities.
Risk Level: Low (doc update of current functionality)
Testing: NA
Docs Changes: Done
Release Notes:
Platform Specific Features: NA
Fixes #25283

### CODE REVIEW

**Comment by nezdolik** [comment]
cc @wbpcode @mattklein123

File: docs/root/intro/arch_overview/upstream/load_balancing/slow_start.rst [hunk 1]
```diff
@@ -61,3 +61,5 @@ Once endpoints E1 and E2 exit slow start mode, their load balancing weight remai
 .. image:: /_static/slow_start_example.svg
    :width: 60%
    :align: center
+
+*Note* that slow start does not work with priorities, e.g. when traffic spills over across priorities there will be no progressive increase of traffic to destination endpoints, all traffic will shift at once.
```
Comment (wbpcode) [inline docs/root/intro/arch_overview/upstream/load_balancing/slow_start.rst:65]: It's hard to say the slow start does not work with priorities. 

**I think the key problems is that the `slow start` can not work for cross-priorities.** It means if there is only one endpoint for a priority, then if the priority is choosed, the unique endpoint is choosed regardless of whether the endpoint in the slow start mode or not.

I am worry about that the current description will make the end users think that if multiple priorities is used then slow start will be invalid completely. But infact, the situation doesn't seem to be that serious.

**Review by wbpcode — COMMENTED** [review]
Thanks for you contribution. A minor comment is added.

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**